### ISSUE TITLE
Error: 'IS0503Test' object has no attribute 'name'

### ISSUE THREAD

**Issue by guojiansheng0925** [AMWA-TV/nmos-testing#717]
Hi,
I tried to run IS-05 controller test suits and got the error written in the title.
The similar situation with #698, but I have set CONFIG.ENABLE_HTTPS = False in the config file.

I have tried both in local environment and docker environment. It is easy to reproduce this issue using docker:
```
docker pull amwa/nmos-testing
docker run -d --network="host" -v="$(pwd)/UserConfig.py:/config/UserConfig.py" amwa/nmos-testing
```
And the config file is simple:
> from . import Config as CONFIG
    CONFIG.ENABLE_HTTPS = False

Then I got the following error:
![image](https://user-images.githubusercontent.com/58507250/194846711-1ccf16c9-f8fa-4494-bc3a-e1e15ff15d59.png)

The config file is attached:
[UserConfig.py.log](https://github.com/AMWA-TV/nmos-testing/files/9746765/UserConfig.py.log)

Any configure/options  should be revised?
Thanks.

**Comment by garethsb** [comment]
This issue, with testing in secure mode, should be resolved by https://github.com/AMWA-TV/nmos-testing/pull/711.

**Comment by garethsb** [comment]
I think it may still be the same issue... if you can try #711, please report the results

**Comment by guojiansheng0925** [comment]
Hi @garethsb , 
thank you for your reply.
I tried #711 and got another error: 
> Error: <nmostesting.TestResult.TestResult object at 0x00000198A04C3280> 

I tried it on my windows laptop and screenshot and the whole log are attached:

![image](https://user-images.githubusercontent.com/58507250/195042793-a345863c-bf70-4ac8-8e32-8fc48e4b42df.png)

[try_PR711.log](https://github.com/AMWA-TV/nmos-testing/files/9753222/try_PR711.log)

Thanks

**Comment by garethsb** [comment]
Thanks, @guojiansheng0925.

@jonathan-r-thorpe Any idea? That message and the log file make it look like the code that handled the first `POST` to the Registration API during the IS-05-03 test setup `raise`d a `TestResult` object rather than wrapping it in an `NMOSTestException`. All of which doesn't make much sense to me...

**Comment by jonathan-r-thorpe** [comment]
@guojiansheng0925 @garethsb Yes there are two problems here:

1. The Testing Tool not contacting the mock Registry successfully
2. Failure to report the error to the user interface correctly

For 1) from the log it appears that the mock resources are being run up on `10.249.172.190`, but the Testing Tool is attempting to  contact it on `192.168.0.106`. From the log:
`POST http://192.168.0.106:5102/x-nmos/registration/v1.3/resource <no response>`

My assumption is that your development machine has more than one network adaptor and the mocks are binding to one adaptor, but the Testing Tool is assuming it has bound to one of the others. This is a bug in the Testing Tool as it should use consistent addresses. (@guojiansheng0925 Is my assumption about the network adaptors correct, or way off the mark?)

And 2) is a separate bug - the error should be propagating the "Registration API returned an unexpected response:..." exception.

I'll see if there is any easy fix for 1), and take a look at 2) when time allows

**Comment by jonathan-r-thorpe** [comment]
@guojiansheng0925 as a temporary work around you could force the Testing Tool to use the correct network adaptor by setting the `BIND_INTERFACE` property in your `UserConfig.py`
```
# Prefer a specific network interface when making mDNS announcements and similar.
# Example: BIND_INTERFACE = 'ens1f0'
CONFIG.BIND_INTERFACE = None
```

**Comment by guojiansheng0925** [comment]
Hi @jonathan-r-thorpe, you are right.
This is caused by my special network environment. 
My laptop connected to a wifi and I used a VPN connection.
192.168.0.106 is the ip of the Wireless LAN adaptor and 10.249.172.190 is the ip of virtual adaptor of VPN.

It works well without the VPN connection.
Thank you very much.

**Comment by garethsb** [comment]
Resolved (possibly only partially) by #711. 🤞
If we still have a (more general) issue when there are multiple network interfaces, I think it would be better to raise that as a new issue.
